invertigo.asl

```diff
@@ -1,106 +1,75 @@
+    vars.Helper.AlertGameTime();
 }
 
 init
 {
-    vars.CoffeeBreakName = "2_coffee_break";
-    vars.MenuName = "menu_v4";
     vars.Helper.TryLoad = (Func<dynamic, bool>)(mono =>
-        {
-            //Thanks to CaptainRektbeard for figuring out the params...
-            vars.Helper["Timer"] = mono.Make<float>("ScoreManager", 1, "_instance", "_timer");
-            vars.Helper["IsTimerRunning"] = mono.Make<bool>("ScoreManager", 1, "_instance", "_running");
-            vars.Helper["IsGamePaused"] = mono.Make<bool>("PauseManager", 1, "_instance", "Paused");
-            return true;
-        });
-    //Init some vars
-    current.PrevLevels = 0.0;
-    current.PrevTimes = 0.0;
-    current.Timer = 0.0;
-    current.Scene = null;
-}
+    {
+        vars.Helper["Timer"] = mono.Make<float>("ScoreManager", 1, "_instance", "_timer");
+        vars.Helper["IsTimerRunning"] = mono.Make<bool>("ScoreManager", 1, "_instance", "_running");
 
-gameTime
-{
-    return TimeSpan.FromSeconds(current.Timer + current.PrevTimes + current.PrevLevels);
+        return true;
+    });
+
+    vars.TotalTime = 0; 
 }
 
 update
 {
-    current.Scene = vars.Helper.Scenes.Active.Name;
-    if (current.Scene == null) {
-        current.Scene = old.Scene;
-    }
-    current.DidReset = old.Timer != null && old.Timer > current.Timer;
-    current.DidLevelChange = current.Scene != old.Scene;
-    if (current.DidReset)
-    {
-        if (current.DidLevelChange)
-        {
-            current.PrevLevels += current.PrevTimes + old.Timer;
-            current.PrevTimes = 0.0;
-        }
-        else
-        {
-            current.PrevTimes += old.Timer;
-        }
-    }
+    current.Scene = vars.Helper.Scenes.Active.Name ?? old.Scene;
 }
 
 start
 {
-    if (current.Scene == vars.CoffeeBreakName && current.IsTimerRunning)
-    {
-        current.PrevLevels = 0.0;
-        current.PrevTimes = 0.0;
-        return true;
-    }
+    return current.Scene == vars.CoffeeBreakName && !old.IsTimerRunning && current.IsTimerRunning;
 }
 
-reset
+onStart
 {
-    if ((settings["resetMenu"] && current.Scene == vars.MenuName)
-        || (settings["resetCoffeeBreak"] && current.Scene == vars.CoffeeBreakName && current.DidReset))
-    {
-        current.PrevLevels = 0.0;
-        current.PrevTimes = 0.0;
-        return true;
-    }
+    vars.TotalTime = 0;
 }
 
 split
 {
-    return current.DidLevelChange;
+    return old.Scene != current.Scene;
+}
+
+reset
+{
+    return settings["resetMenu"] && old.Scene != current.Scene && current.Scene == vars.MenuName
+        || settings["resetCoffeeBreak"] && current.Scene == vars.CoffeeBreakName && old.Timer > current.Timer;
+}
+
+gameTime
+{
+    if (old.Timer > current.Timer)
+    {
+        vars.TotalTime += old.Timer - current.Timer;
+    }
+
+    return TimeSpan.FromSeconds(vars.TotalTime + current.Timer);
 }
 
 isLoading
 {
-    return !current.IsTimerRunning || current.IsGamePaused;
-}
\ No newline at end of file
+    return true;
```

Comment:
When using `gameTime`, you should *always* return a constant value of `true` in `isLoading`.